app/assets/javascripts/assignment_swapping.js

```diff
@@ -0,0 +1,57 @@
+function add_events()
+{
+  var form = document.getElementById("swapping_form");
+
+  if(form != null)
+  {
+   form.addEventListener("submit", function(event){ check_valid_form(event); })
+  }
+
+  var input_elements = document.getElementsByTagName("input");
+
+  for(var i = 0; i<input_elements.length; i++)
```

Comment:
`for ... of` should work here. New school JS :)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of